vega_sim/scenario/fuzzed_markets/agents.py

```diff
@@ -0,0 +1,347 @@
+                key_name=self.key_name,
+                asset=self.asset_id,
+                amount=self.initial_asset_mint,
+                wallet_name=self.wallet_name,
+            )
+
+        self.vega.wait_fn(5)
+
+    def step(self, vega_state):
+        self.live_orders = self.vega.orders_for_party_from_feed(
+            key_name=self.key_name,
+            wallet_name=self.wallet_name,
+            market_id=self.market_id,
+            live_only=True,
+        )
+
+        self.curr_price = vega_state.market_state[self.market_id].midprice
+
+        submissions = [self.create_fuzzed_submission() for _ in range(20)]
+        amendments = [self.create_fuzzed_amendment() for _ in range(10)]
+        cancellations = [self.create_fuzzed_cancellation() for _ in range(1)]
+
+        self.vega.submit_instructions(
+            key_name=self.key_name,
+            wallet_name=self.wallet_name,
+            submissions=submissions,
+            amendments=amendments,
+            cancellations=cancellations,
+        )
+
+    def create_fuzzed_cancellation(self):
+        if self.live_orders != {}:
+            order_key = self.random_state.choice(list(self.live_orders.keys()))
+            order_id = self.live_orders[order_key].id
+        else:
+            order_id = None
+
+        return self.vega.create_order_cancellation(
+            order_id=order_id,
+            market_id=self.market_id,
+        )
+
+    def create_fuzzed_amendment(self):
+        if self.live_orders != {}:
+            order_key = self.random_state.choice(list(self.live_orders.keys()))
+            order_id = self.live_orders[order_key].id
+        else:
+            order_id = None
+
+        return self.vega.create_order_amendment(
+            order_id=order_id,
+            market_id=self.market_id,
+            size_delta=self.random_state.normal(loc=0, scale=5),
+            time_in_force=self.random_state.choice(
+                a=[
+                    "TIME_IN_FORCE_UNSPECIFIED",
+                    "TIME_IN_FORCE_GTC",
+                    "TIME_IN_FORCE_GTT",
+                    "TIME_IN_FORCE_GFN",
+                    "TIME_IN_FORCE_GFA",
+                    "TIME_IN_FORCE_FOK",
+                    "TIME_IN_FORCE_IOC",
+                ]
+            ),
+            price=self.random_state.choice(
+                a=[None, self.random_state.normal(loc=self.curr_price, scale=20)]
+            ),
+            expires_at=int(
+                (
+                    self.vega.get_blockchain_time()
+                    + self.random_state.normal(loc=60, scale=60)
+                )
+                * 1e9
+            ),
+            pegged_reference=self.random_state.choice(
+                a=[
+                    "PEGGED_REFERENCE_UNSPECIFIED",
+                    "PEGGED_REFERENCE_MID",
+                    "PEGGED_REFERENCE_BEST_BID",
+                    "PEGGED_REFERENCE_BEST_ASK",
+                ]
+            ),
```

Comment:
Can we tweak the probabilities on these so that `UNSPECIFIED` is chosen as often as the other three combined? Will mean we get vaguely 50:50 pegged vs unpegged